src/pushsource/_impl/model/ami.py

```diff
@@ -92,6 +92,27 @@ def _from_data(cls, data):
         return cls(**kwargs)
 
 
+@attr.s()
+class AmiAccessEndpointUrl(object):
+    """Access Endpoint URL to be associated with a Marketplace VM."""
+
+    port = attr.ib(type=int, validator=instance_of(int))
+    """Port to access the endpoint URL."""
+
+    protocol = attr.ib(type=str, validator=instance_of_str)
```

Comment:
Minor: I'm not sure if you are copying some existing AWS API which already uses the term "protocol".  If so, ignore this.

But in the context of a URL, this part of a URL is referred to as a "scheme", not "protocol".  For example the named tuple returned by https://docs.python.org/3/library/urllib.parse.html uses "scheme".  Given that this class also represents a URL, it would be nice to be consistent with that.